spec/checkPullRequestReviewSpec.js

```diff
@@ -1000,7 +1023,269 @@ describe('Pull Request Review Module', () => {
     });
 
     afterAll(() => {
-      payloadData.payload.pull_request.state = initalState;
+      reviewPayloadData.payload.pull_request.state = initalState;
+    });
+  });
+
+  describe('Pull request author comments on PR', () => {
+    beforeEach(() => {
+      github.pulls = {
+        get: jasmine.createSpy('get').and.resolveTo({
+          data: commentPayloadData.payload.issue,
+        }),
+      };
+    });
+
+    describe(
+      'when author asks reviewer to ptal and does not assign them', () => {
+        const initialAssignees = commentPayloadData.payload.issue.assignees;
+        beforeAll(() => {
+          commentPayloadData.payload.issue.assignees = [];
+        });
+        beforeEach(async () => {
+          await robot.receive(commentPayloadData);
+        });
+
+        it('should check call response to review method', () => {
+          expect(
+            pullRequestReviewModule.handleResponseToReview
+          ).toHaveBeenCalled();
+        });
+
+        it('should wait for 3 minutes before performing any action', () => {
+          expect(utilityModule.sleep).toHaveBeenCalled();
+          expect(utilityModule.sleep).toHaveBeenCalledWith(THREE_MINUTES);
+        });
+
+        it('should get the updated version of the pull request', () => {
+          expect(github.pulls.get).toHaveBeenCalled();
+          expect(github.pulls.get).toHaveBeenCalledWith({
+            repo: commentPayloadData.payload.repository.name,
+            owner: commentPayloadData.payload.repository.owner.login,
+            pull_number: commentPayloadData.payload.issue.number,
+          });
+        });
+
+        it('should assign reviewers', () => {
+          expect(github.issues.addAssignees).toHaveBeenCalled();
+          expect(github.issues.addAssignees).toHaveBeenCalledWith({
+            repo: commentPayloadData.payload.repository.name,
+            owner: commentPayloadData.payload.repository.owner.login,
+            issue_number: commentPayloadData.payload.issue.number,
+            assignees: ['reviewer1', 'reviewer2'],
+          });
+        });
+
+        afterAll(() => {
+          commentPayloadData.payload.issue.assignees = initialAssignees;
+        });
+      }
+    );
+
+    describe(
+      'when author asks reviewer to ptal and assigns them already', () => {
```

Comment:
One more case - assigns some of them. Also, are we unassigning the author?

Reply:
No, we are not unassigning the author. There's a comment from Ben on the doc outlining why.

Reply:
https://docs.google.com/document/d/1m-7dZWGw77hZ6GOnjBu4p1XGXxyBPY2n2pzf-76RuTY/edit?disco=AAAAJ4jo7Es

Reply:
Replied there as well.